src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts

```diff
@@ -112,9 +112,11 @@ export class McqQuestionEditDetailsFormComponent
   triggerWeightsColumn(event: any): void {
     const fieldsToUpdate: any = {};
     if (!event.target.checked) {
+      this.model.hasAssignedWeights = false;
       fieldsToUpdate.mcqWeights = [];
       fieldsToUpdate.mcqOtherWeight = 0;
     } else {
+      this.model.hasAssignedWeights = true;
```

Comment:
By doing this, you are still changing the data that belongs to the parent in the child component.